### PR TITLE
CMR-4474

### PR DESCRIPTION
**Pull request by ygliuvt**
As a metadata-db user, I want to search service association concepts by service-concept-id or associated-concept-id.

### CODE REVIEW

File: metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/service_association_search_test.clj [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                                                                 :short-name "s2"}})
+        associated-service (util/create-and-save-service "REG_PROV" 1)
+        serv-association1 (util/create-and-save-service-association coll1 associated-service 1 3)
+        serv-association2 (util/create-and-save-service-association coll2 associated-service 2 2)]
+    (testing "find latest revisions"
+      (are3 [service-associations params]
+        (is (= (set service-associations)
+               (set (->> (util/find-latest-concepts :service-association params)
+                         :concepts
+                         (map #(dissoc % :provider-id :revision-date :transaction-id))))))
```
Comment (chris-durbin) [inline metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/service_association_search_test.clj]: Add :created-at to this.

File: metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/tag_search_test.clj [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+(deftest find-tags
+  (let [tag1 (util/create-and-save-tag 1 3)
+        tag2 (util/create-and-save-tag 2 2)]
+    (testing "find latest revsions"
+      (are3 [tags params]
+        (is (= (set tags)
+               (set (->> (util/find-latest-concepts :tag params)
+                         :concepts
+                         (map #(dissoc % :provider-id :revision-date :transaction-id))))))
```
Comment (chris-durbin) [inline metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/tag_search_test.clj]: :created-at as well

File: metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/variable_association_search_test.clj [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                                                                 :short-name "s2"}})
+        associated-variable (util/create-and-save-variable "REG_PROV" 1)
+        var-association1 (util/create-and-save-variable-association coll1 associated-variable 1 3)
+        var-association2 (util/create-and-save-variable-association coll2 associated-variable 2 2)]
+    (testing "find latest revisions"
+      (are3 [variable-associations params]
+        (is (= (set variable-associations)
+               (set (->> (util/find-latest-concepts :variable-association params)
+                         :concepts
+                         (map #(dissoc % :provider-id :revision-date :transaction-id))))))
```
Comment (chris-durbin) [inline metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/variable_association_search_test.clj]: :created-at...

File: metadata-db-app/int_test/cmr/metadata_db/int_test/utility.clj [hunk 4]
```diff
@@ -653,6 +653,18 @@
     (is (= 200 status))
     (is (= deleted? (:deleted concept)))))
 
+(defn concepts-for-comparison
+  "Removes irrelevant fields from concepts so they can be compared in search tests."
+  [concepts]
+  (map #(dissoc % :revision-date :transaction-id :created-at) concepts))
```
Comment (chris-durbin) [inline metadata-db-app/int_test/cmr/metadata_db/int_test/utility.clj]: Yeah for this one.

File: metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/service_association_search_test.clj (hunk 1 above)
Comment (ygliuvt) [reply metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/search/service_association_search_test.clj]: I would rather do this later after this PR  is merged as this branch is building fine.

**Pull request merged**